src/OBIUtils.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+export interface LGItem {
+    text: string,
+    suggestions: string[]
+}
+
+const findOrThrow = (haystack_fn: (s: string) => number, needle: string): number => {
+    const i = haystack_fn(needle)
+    if (i === -1) throw new RangeError(`${needle} not found`)
+    return i
+}
+
+/**
+ * Finds the index of the first instance of needle in haystack, or throws an exception if not found.
+ */
+const findFirstOrThrow = (haystack: string, needle: string): number => {
+    return findOrThrow(haystack.indexOf.bind(haystack), needle)
+}
+
+/**
+ * Finds the index of the last instance of needle in haystack, or throws an exception if not found.
+ */
+const findLastOrThrow = (haystack: string, needle: string): number => {
+    return findOrThrow(haystack.lastIndexOf.bind(haystack), needle)
+}
+
+/**
+ * Parses OBI .dialog files from custom format into structured data.
+ */
+export class ObiUtils {
+
+    public static parseLGString(text: string): Map<string, LGItem> {
```

Comment:
One difference with this version is that it creates a new Map.  The previous version appends to an existing map.  In the sample you have there's only on LG file, but there could me many

Reply:
Is that a problem ?  It's reasonably straightforward to create the union of several Maps

`const union = new Map([...a, ...b, ...c])`

Happy to change it to take the Map parameter if that seems more straightforward.

Reply:
Computationally it more intense, but fine.  :)

Reply:
Actually, my proposal doesn't work with an array of Maps.  Still learning where the rough edges of Typescript are.

Updated to take input Map param.